core/src/main/scala/cats/FlatMap.scala

```diff
@@ -89,6 +89,18 @@ import simulacrum.noop
   override def product[A, B](fa: F[A], fb: F[B]): F[(A, B)] =
     flatMap(fa)(a => map(fb)(b => (a, b)))
 
+  override def ap2[A, B, Z](ff: F[(A, B) => Z])(fa: F[A], fb: F[B]): F[Z] =
+    flatMap(fa)(a => flatMap(fb)(b => map(ff)(_(a, b))))
+
+  override def map2[A, B, Z](fa: F[A], fb: F[B])(f: (A, B) => Z): F[Z] =
+    flatMap(fa)(a => map(fb)(b => f(a, b)))
+
+  override def productR[A, B](fa: F[A])(fb: F[B]): F[B] =
+    flatMap(fa)(_ => fb)
```

Comment:
Coming back to this, the lack of symmetry between `productR` and `productL` feels a little off to me. I think that the override of `map2` makes sense. But maybe `productL` and `productR` should both just delegate to `map2`?

Reply:
Unlike `productL`, `productR` using `map2` will incur one extra `map`. That's where the asymmetry comes from.

Reply:
Oh right. Sorry that I've required so much explanation on this one 😬

LGTM. @mpilquist does this PR look good to you?

Reply:
Yep looks good to me